sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriterV2.scala

```diff
@@ -158,7 +158,9 @@ final class DataFrameWriterV2[T] private[sql](table: String, ds: Dataset[T])
   def append(): Unit = {
     val append = loadTable(catalog, identifier) match {
       case Some(t) =>
-        AppendData.byName(DataSourceV2Relation.create(t), logicalPlan, options.toMap)
+        AppendData.byName(
+          DataSourceV2Relation.create(t, Some(catalog), Seq(identifier)),
```

Comment:
we probably need to use a complex extractor above to grab which catalog was used if any